rm-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/rm/client/nodesource/NodeSourceWindow.java

```diff
@@ -685,14 +673,35 @@ private void populateHiddenItemsIfNecessary(PluginDescriptor pluginDescriptor) {
         return allFormItems;
     }
 
-    private StaticTextItem createElasticLabel(PluginDescriptor plugin) {
-        StaticTextItem elasticInfra = new StaticTextItem(plugin.getPluginName() + "elastic0");
-        elasticInfra.setTitle("");
-        elasticInfra.setValue("Elastic Infrastructure");
-        FormItemIcon icon = new FormItemIcon();
-        icon.setSrc(RMImages.instance.good().getSafeUri().asString());
-        elasticInfra.setIcons(icon);
-        return elasticInfra;
+    private void addElasticLabelIfNecessary(PluginDescriptor plugin, List<FormItem> allFormItems) {
+        if ("true".equalsIgnoreCase(plugin.getMeta().get("elastic"))) {
+            StaticTextItem elasticInfra = new StaticTextItem(plugin.getPluginName() + "elastic0");
+            elasticInfra.setTitle("");
+            elasticInfra.setTitleStyle("important-message");
+            elasticInfra.setTextBoxStyle("grey-message");
+            elasticInfra.setValue("Elastic");
+
+            FormItemIcon icon = new FormItemIcon();
+            icon.setSrc(RMImages.instance.good().getSafeUri().asString());
+            elasticInfra.setIcons(icon);
+            allFormItems.add(elasticInfra);
+        }
+    }
+
+    private String orderedFields(PluginDescriptor plugin) {
```

Comment:
func can be renamed as `orderedNonFileFields` ?

Reply:
ok thanks